doc/getting-started/generating-keys.md

```diff
@@ -1,10 +1,97 @@
+
+This is demonstrated with the below code
+```javascript
+function harden(num: number): number {
+  return 0x80000000 + num;
+}
+
+
+const rootKey = CardanoWasm.BIP32PrivateKey.from_bech32("xprv17qx9vxm6060qjn5fgazfue9nwyf448w7upk60c3epln82vumg9r9kxzsud9uv5rfscxp382j2aku254zj3qfx9fx39t6hjwtmwq85uunsd8x0st3j66lzf5yn30hwq5n75zeuplepx8vxc502txx09ygjgx06n0p");
+const accountKey = rootKey
+  .derive(harden(1852)) // purpose
+  .derive(harden(1815)) // coin type
+  .derive(harden(0)); // account #0
+
+const utxoPubKey = accountKey
+  .derive(0) // external
+  .derive(0)
+  .to_public();
+
+const stakeKey = accountKey
+  .derive(2) // chimeric
+  .derive(0)
+  .to_public();
+```
+
+## BIP39 Entropy
+
+To generate a `BIP32PrivateKey` from a BIP39 recovery phrase it must be first converted to entropy following the [BIP39 protocol](). This library does not directly handle that, but once entropy is created it is possible to use `Bip32PrivateKey.from_bip39_entropy(entropy, password)`. For more information see the [CIP3](https://github.com/cardano-foundation/CIPs/pull/3) Cardano improvement proposal. The code below uses the `bip39` npm package to generate a root `BIP32PrivateKey` from a BIP39 mnemonic.
+
+```javascript
+import { mnemonicToEntropy } from 'bip39';
+
+const entropy = mnemonicToEntropy(
+  [ "test", "walk", "nut", "penalty", "hip", "pave", "soap", "entry", "language", "right", "filter", "choice" ].join(' ')
+);
+
+const rootKey = CardanoWasm.Bip32PrivateKey.from_bip39_entropy(
+  Buffer.from(entropy, 'hex'),
+  Buffer.from(''),
+);
+```
+
+## Use in Addresses
+
+Once we have reached the desired derivation path, we must convert the `BIP32PrivateKey` or `BIP32PublicKey` to a `PrivateKey` or `PublicKey` by calling `.to_raw_key()` on them.
+For example, to create an address using the `utxoPubKey` and `stakeKey` in the first example, we can do:
+```javascript
+// network id - 0? for mainnet, 1? for testnet
+const networkId = 0;
+
+// base address with staking key
+const baseAddr = CardanoWasm.BaseAddress.new(
+  networkId,
+  CardanoWasm.StakeCredential.from_keyhash(utxoPubKey.to_raw_key().hash()),
+  CardanoWasm.StakeCredential.from_keyhash(stakeKey.to_raw_key().hash()),
+);
+
+// enterprise address without staking ability, for use by exchanges/etc
+const enterpriseAddr = CardanoWasm.EnterpriseAddress.new(
+  networkId,
+  CardanoWasm.StakeCredential.from_keyhash(utxoPubKey.to_raw_key().hash())
+);
+
+// pointer address - similar to Base address but can be shorter, see formal spec for explanation
+const ptrAddr = CardanoWasm.PointerAddress.new(
+  networkId,
+  CardanoWasm.StakeCredential.from_keyhash(utxoPubKey.to_raw_key().hash()),
+  CardanoWasm.Pointer.new(
+    100, // slot
+    2,   // tx index in slot
+    0    // cert indiex in tx
+  )
+);
+```
+
+These are all address variant types with information specific to its address type. To be used with the library we must conver them to a generic `Address` like such:
```

Comment:
I'm not sure what this means. What about Reward address and Byron address?

Notably, I would mention you can use `from_icarus_key` to generate an Ae2 (Icarus-style) Byron address from a key, but our library doesn't support generated Dd (Daedalus-style) Byron addresses.